### DISCUSSION TITLE
Lighthouse Not Respecting User Table Name

### DISCUSSION THREAD

**Discussion by JediKev** [nuwave/lighthouse#2536] [Q&A] [answered]
Hello Everyone,

I have read through the documentation, many guides online, SO, etc. and I cannot for the life of me figure out what's going on in my instance. It's very possible that I've missed something or skipped over something so I apologize in advance if that's the case. To preface I'm using Laravel 9 (will be upgrading soon) and Lighthouse v6.36.0.

My User table name is simply `user` (singular; not plural) which deviates from core Laravel. In all aspects of the application the `user` table name has _zero_ issues. However, with Lighthouse, when I run a simple User query to retrieve all Users (or a specific User by ID) it gives the following error:
```
Base table or view not found: 1146 Table 'database_name.users' doesn't exist
```

Just to note I do have `protected $table = 'user';` defined in the User class (again, everywhere else but lighthouse works fine). Additionally, I have the `'driver' => 'eloquent'` with `'model' => App\Models\User::class` defined in my `config/auth.php` file. I also cleared all artisan cache/config/etc. and even cleared lighthouse cache but still no dice. If I run a simple query for any other Model it returns the expected results. The only table having issues is the User table.

Any help or suggestions would be greatly appreciated. Lastly, if I've missed any important information please let me know.

Cheers.

**Comment by spawnia** [comment]
Do other queries work, or is it specific to querying users?

**Reply by JediKev** [reply]
Just the User table:
>If I run a simple query for any other Model it returns the expected results. The only table having issues is the User table.

**Reply by spawnia** [reply]
How does the schema for your User queries look like?

**Reply by JediKev** [reply, marked as answer]
Ain't no way 💀 I'm just dumb. I had a duplicate User class file under the main `app/` directory. It was left over from when initializing the project long ago. Once I removed the dupe file it's working as expected!

**Comment by JediKev** [comment]
Update, after sifting through the endless stack trace I wound up nowhere. So I took a step back and did some research on singular table names within Laravel itself with no mention of Lighthouse/GraphQL. Looking through the search results I came across the `Illuminate\Support\Pluralizer` class. I noticed it had a property called `$uncountable` containing strings to avoid when pluralizing. So as a quick test I added `'user'` to the `$uncountable` array and voila the issue was no more! BUT this should not be the accepted solution as you would have to modify core every time.

So my next question would be why does the entire Laravel application respect the `protected $table = 'user';` definition in _all_ regards (Authentication, Eloquent ORM, etc.) but Lighthouse does not? Lighthouse appears to be bypassing the Eloquent norms of grabbing the true table name from the `$table` property of the Model class. If the Object Types relate to the Models, why doesn't Lighthouse call `Model->getTable()` to get the true table name? Do I need to add my own Mutation/Query resolver specifically for the User table or something?
